viahtml/blocklist.py

```diff
@@ -0,0 +1,116 @@
+"""The abstract blocklist object."""
+
+import os
+import re
+from enum import Enum
+from logging import getLogger
+from urllib.parse import urlparse
+
+
+class Blocklist:
+    """A blocklist which URLs can be checked against.
+
+    The blocklist is stored at:
+      * https://hypothesis-via.s3-us-west-1.amazonaws.com/via-blocklist.txt
+
+    And is downloaded locally, on a loop by a cron job.
+    """
+
+    LOG = getLogger(__name__)
+
+    class Reason(Enum):
+        """List of reasons a blocklist can be blocked."""
+
+        PUBLISHER_BLOCKED = "publisher-blocked"  # Content owner has asked us to block
+        MEDIA_VIDEO = "media-video"  # Sites which are mostly video content
+        MEDIA_AUDIO = "media-audio"  # Sites which are mostly audio content
+        MEDIA_IMAGE = "media-image"  # Sites which are mostly image content
+        MEDIA_MIXED = "media-mixed"  # Sites with a mixture of content
+        HIGH_IO = "high-io"  # Sites with high interactivity and AJAX calls
+        OTHER = "other"  # Escape hatch for poorly formatted values
+
+        @classmethod
+        def parse(cls, value):
+            """Parse a value into an enum object."""
+            try:
+                return cls(value)
+            except ValueError:
+                return cls.OTHER
+
+    # viahtml is ok with video, as far as we can tell
+    PERMITTED = (Reason.MEDIA_VIDEO,)
```

Comment:
Yeah, this isn't true, but this is where you'd disable this without changing the block file